### PR TITLE
394 implement refresh token

### PR DESCRIPTION
**Pull request by anncatton**
- cleans up logic for expired refresh token
- cleans up comments

### CODE REVIEW

File: src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java [hunk 1]
```diff
@@ -36,11 +37,34 @@
   @Rule public ExpectedException exceptionRule = ExpectedException.none();
 
   @Test
-  public void refresh_usedTokenIsDeleted() {
+  public void refresh_invalidContext_usedTokenIsDeleted() {
+    val user1 = entityGenerator.setupUserWithRefreshToken("User One");
+    // force invalid refreshContext with invalid user status
+    user1.setStatus(StatusType.PENDING);
```
Comment (rtisma) [inline src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java]: Our general guideline for static methods/fields is to statically import them. Intellij has an option that can do the static import for you

File: src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java [hunk 2]
```diff
@@ -36,11 +37,34 @@
...
+    val user1Token = tokenService.generateUserToken(user1);
+    val refreshToken1 = user1.getRefreshToken();
+
+    Assert.assertTrue(refreshContextService.getById(refreshToken1.getId()) != null);
```
Comment (rtisma) [inline src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java]: this assertion is not enough. A better one is:
```java
assertTrue(refreshContextService.getById(refreshToken1.getId) != null);
assertTrue(refreshContextService.getById(refreshToken1.getId()).isPresent());
```
Even just doing the isPresent check is enough for a test. In general, if an `Optional<>` object is returned then you can assume the developer intends it to always be not null. you can have empty optional objects, and non-empty optional objects. Returning null on top of that defeats the purpose of Optional :)

File: src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java [hunk 3]
```diff
@@ -36,11 +37,34 @@
...
+    val incomingRefreshContext =
+        refreshContextService.createRefreshContext(refreshToken1.getId().toString(), user1Token);
+    refreshContextService.disassociateUserAndDelete(user1Token);
+
+    exceptionRule.expect(NotFoundException.class);
+    exceptionRule.expectMessage(
+        String.format("RefreshToken '%s' does not exist", refreshToken1.getId()));
+    Assert.assertTrue(refreshContextService.get(refreshToken1.getId(), false) == refreshToken1);
```
Comment (rtisma) [inline src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java]: this is a java gotcha! Read below
https://stackoverflow.com/a/10806192/1952450

basically, you have to use `.equals()` for this type of comparison

In a nut shell, the following code runs without failure/errors:
```java
RefreshToken r1 = RefreshToken.builder().id("someId").build();
RefreshToken r2 = r1;
RefreshToken r3 = RefreshToken.builder().id("someId").build();
RefreshToken r4 = RefreshToken.builder().id("SOMETHING_ELSE").build();

assertTrue(r1 == r2);
assertTrue(r1 != r3);
assertTrue(r1 != r4);
assertTrue(r2 != r3);
assertTrue(r2 != r4); 
assertTrue(r3 != r4);

assertTrue(r1.equals(r2));
assertTrue(r1.equals(r3));
assertFalse(r1.equals(r4));
assertTrue(r2.equals(r3));
assertFalse(r2.equals(r4));
assertFalse(r3.equals(r4));
```

For the selected line of code, you can use:
```java
assertEquals(refreshContextService.get(refreshToken1.getId(), false), refreshToken1);
```
which calls the .equals method under the hood.

Comment (anncatton) [reply src/test/java/bio/overture/ego/service/RefreshContextServiceTest.java]: Aha...i ran into this with a string comparison but didn't realize it here!

**Comment by anncatton** [comment]
ohh yes forgot about the assert imports - will do that before merging!

**Pull request merged**